### PR TITLE
Moving generated Go protobuf files to match go_package

### PR DESCRIPTION
**Pull request by jhwillett**
Evaluating whether, by rearranging https://github.com/livekit/protocol, we can produce `*.proto` and `*.pb.go` pairs which can both be imported from proto files in other projects.

### CODE REVIEW

File: magefile.go [hunk 1]
```diff
@@ -9,36 +9,23 @@ import (
 	"os"
 	"os/exec"
 	"path/filepath"
-
-	"github.com/magefile/mage/target"
```
Comment (jhwillett) [inline magefile.go:13]: This seems to have been removed by `goimports`.

File: magefile.go [hunk 2]
```diff
@@ -9,36 +9,23 @@ import (
...
-	)
-	if err != nil {
-		return err
+	twirpProtoFiles := []string{
+		"proto/livekit_recording.proto",
+		"proto/livekit_room.proto",
 	}
-	if !updated {
-		return nil
-	}
```
Comment (jhwillett) [inline magefile.go:34]: I have removed the dirtyness check.  We do not practice it in other projects for either wire nor protobuf generation, and it feels dangerous.

I did add more explicit lists of source files.

File: proto/livekit_analytics.pb.go [hunk 3]
```diff
@@ -2,9 +2,9 @@
 // versions:
 // 	protoc-gen-go v1.27.1
 // 	protoc        v3.17.3
-// source: livekit_analytics.proto
+// source: proto/livekit_analytics.proto
 
-package livekit
+package proto
```
Comment (jhwillett) [inline proto/livekit_analytics.pb.go:7]: We have a gotcha: an unfortunate trade-off.

With `--go_out . --go_opt=paths=source_relative`, which is the only pattern that puts `livekit_*.pb.go` in this location, the package name must match the directory name.

That is, if we wish to have Go files at `proto/livekit_*.pb.go`, they must have `package proto`.

If we wish them to have `package livekit`, we must indicate `go_package = something/livekit` in the `*.proto` files and then we will end up with `livekit/livekit_*.pb.go`.

Either way we will break some kind of backward compatibility.  Either the `livekit_*.pb.go` package names must change, or their paths must change.

Comment (jhwillett) [reply proto/livekit_analytics.pb.go:7]: IMO, of these alternatives the prettier one is to end up with `package livekit` and to put all the files in `livekit/livekit_*.pb.go`.

However, before changing it again I will wait for a discussion and the conclusion of my test in another project, of importing these types.

**Review by frostbyte73 — APPROVED** [review]

**Pull request merged**